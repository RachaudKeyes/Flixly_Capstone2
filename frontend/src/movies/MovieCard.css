
.MovieCardBody {
  background-color: rgb(54, 0, 0);
  width: 12rem;
  height: 24rem;
  margin-bottom: 1rem;
  color: white;
  border-radius: 0;
  border: none;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
}

.MovieCard img {
  width: 12rem;
}

.MovieCard:hover { /* <Card /> that is also <Link /> */
  text-decoration: none;
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
}
