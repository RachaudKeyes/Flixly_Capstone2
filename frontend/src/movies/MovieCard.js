import React from "react";
import { Link } from "react-router-dom";
import "./MovieCard.css";

/** Show limited information about a movie
 *
 * Is rendered by MovieList and FavoriteList to show a "card" for each movie.
 *
 * { MovieList, FavoriteList } -> MovieCard
 */

function MovieCard( { id, title, releaseDate, poster, voteAverage}  ) {
    console.debug("MovieCard");

    return (
      <div className="MovieCardBody">
        <Link className="MovieCard" to={`/movies/${id}`}>
          <div>
            { poster && <img src={`http://image.tmdb.org/t/p/w500/${poster}`}
                              alt={ title }
                              className="" /> }
          </div>
        </Link>
        <div className="font-weight-bold mb-4"> { title } </div> 
        <div className="mt-3 mb-0">
          { releaseDate }
          <span className="float-right">{ Math.round(voteAverage) }%</span>
        </div>     
      </div> 
    );
}

export default MovieCard;