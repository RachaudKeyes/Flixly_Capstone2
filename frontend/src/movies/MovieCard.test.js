import React from "react";
import { render } from "@testing-library/react";
import MovieCard from "./MovieCard";
import { MemoryRouter } from "react-router";

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <MovieCard
            key={ 668489 }
            id= { 668489 }
            title= "Havoc"
            releaseDate={ 2025 }
            poster= "/r46leE6PSzLR3pnVzaxx5Q30yUF.jpg"
            voteAverage={ 6.608 * 10 }
        />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

