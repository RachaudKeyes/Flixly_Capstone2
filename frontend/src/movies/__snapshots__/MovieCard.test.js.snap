// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MovieCardBody"
  >
    <a
      class="MovieCard"
      href="/movies/668489"
    >
      <div>
        <img
          alt="Havoc"
          class=""
          src="http://image.tmdb.org/t/p/w500//r46leE6PSzLR3pnVzaxx5Q30yUF.jpg"
        />
      </div>
    </a>
    <div
      class="font-weight-bold mb-4"
    >
       Havoc 
    </div>
    <div
      class="mt-3 mb-0"
    >
      2025
      <span
        class="float-right"
      >
        66%
      </span>
    </div>
  </div>
</DocumentFragment>
`;
